### PR TITLE
Added math function power

### PR DESCRIPTION
**Pull request by shellhub**
#### Checklist

- [ ] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/604"></a>

### CODE REVIEW

File: misc/power.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        result *= a;
+    }
+    return result;
+}
+
+/**
+ * Test function
+ * @return void
+ */
+void test()
```
Comment (Panquesito7) [inline misc/power.c:29]:
```suggestion
static void test()
```

File: misc/power_recursion.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * @param b the exponent.
+ * @return the value {@code a}<sup>{@code b}</sup>.
+ */
+long power(int a, int b) { return b == 0 ? 1 : a * power(a, b - 1); }
+
+/**
+ * Test function
+ * @return void
+ */
+void test()
```
Comment (Panquesito7) [inline misc/power_recursion.c:22]:
```suggestion
static void test()
```

File: misc/power_recursion.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+{
+    assert(power(0, 2) == 0);
+    assert(power(2, 3) == 8);
+    assert(power(2, 10) == 1024);
+    assert(power(3, 3) == 27);
+}
+
+/**
+ * Driver Code
+ * @return None
```
Comment (Panquesito7) [inline misc/power_recursion.c:32]:
```suggestion
 * @returns 0 on exit
```

File: misc/power.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+{
+    assert(power(0, 2) == 0);
+    assert(power(2, 3) == 8);
+    assert(power(2, 10) == 1024);
+    assert(power(3, 3) == 27);
+}
+
+/**
+ * Driver Code
+ * @return None
```
Comment (Panquesito7) [inline misc/power.c:39]:
```suggestion
 * @returns 0 on exit
```

File: misc/power.c [hunk 5]
```diff
@@ -0,0 +1,45 @@
+/**
+ * @file
+ * @brief Program to calculate
+ * [exponentiation](https://en.wikipedia.org/wiki/Exponentiation)
```
Comment (Panquesito7) [inline misc/power.c:4]:
```suggestion
 * [exponentiation](https://en.wikipedia.org/wiki/Exponentiation)
 *
 * @author [Du Yuanchao](https://github.com/shellhub)
```

File: misc/power_recursion.c [hunk 6]
```diff
@@ -0,0 +1,38 @@
+/**
+ * @file
+ * @brief Program to calculate
+ * [exponentiation](https://en.wikipedia.org/wiki/Exponentiation) using
+ * recursion algorithm.
```
Comment (Panquesito7) [inline misc/power_recursion.c:5]:
```suggestion
 * recursion algorithm.
 *
 * @author [Du Yuanchao](https://github.com/shellhub)
```

File: misc/power.c (hunk 5 above)
Comment (shellhub) [reply misc/power.c:4]: I think author is optional.

File: misc/power.c [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    long result = 1;
+    for (int i = 1; i <= b; ++i)
+    {
+        result *= a;
+    }
+    return result;
+}
+
+/**
+ * Test function
```
Comment (Panquesito7) [inline misc/power.c:28]:
```suggestion
 * @brief Test function
```

File: misc/power.c [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+static void test()
+{
+    assert(power(0, 2) == 0);
+    assert(power(2, 3) == 8);
+    assert(power(2, 10) == 1024);
+    assert(power(3, 3) == 27);
+}
+
+/**
+ * Driver Code
```
Comment (Panquesito7) [inline misc/power.c:40]:
```suggestion
 * @brief Driver Code
```

File: misc/power_recursion.c [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+static void test()
+{
+    assert(power(0, 2) == 0);
+    assert(power(2, 3) == 8);
+    assert(power(2, 10) == 1024);
+    assert(power(3, 3) == 27);
+}
+
+/**
+ * Driver Code
```
Comment (Panquesito7) [inline misc/power_recursion.c:31]:
```suggestion
 * @brief Driver Code
```

File: misc/power_recursion.c [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+ * Returns the value of the first argument raised to the power of the
+ * second argument using recursion.
+ * @param a the base.
+ * @param b the exponent.
+ * @return the value {@code a}<sup>{@code b}</sup>.
+ */
+long power(int a, int b) { return b == 0 ? 1 : a * power(a, b - 1); }
+
+/**
+ * Test function
```
Comment (Panquesito7) [inline misc/power_recursion.c:19]:
```suggestion
 * @brief Test function
```

File: misc/power.c [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+ * @file
+ * @brief Program to calculate
+ * [exponentiation](https://en.wikipedia.org/wiki/Exponentiation)
+ *
+ * @author [Du Yuanchao](https://github.com/shellhub)
+ */
+#include <assert.h>
+
+/**
+ * Returns the value of the first argument raised to the power of the
```
Comment (Panquesito7) [inline misc/power.c:11]:
```suggestion
 * @brief Returns the value of the first argument raised to the power of the
```

File: misc/power_recursion.c [hunk 12]
```diff
@@ -0,0 +1,38 @@
+/**
+ * @file
+ * @brief Program to calculate
+ * [exponentiation](https://en.wikipedia.org/wiki/Exponentiation) using
+ * recursion algorithm.
+ */
+#include <assert.h>
+
+/**
+ * Returns the value of the first argument raised to the power of the
```
Comment (Panquesito7) [inline misc/power_recursion.c:10]:
```suggestion
 * @brief Returns the value of the first argument raised to the power of the
```

File: misc/power_recursion.c [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+ * second argument using recursion.
+ * @param a the base.
+ * @param b the exponent.
+ * @return the value {@code a}<sup>{@code b}</sup>.
```
Comment (Panquesito7) [inline misc/power_recursion.c:14]:
```suggestion
 * @returns the value {@code a}<sup>{@code b}</sup>.
```

File: misc/power.c [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+ * second argument.
+ * @param a the base.
+ * @param b the exponent.
+ * @return the value {@code a}<sup>{@code b}</sup>.
```
Comment (Panquesito7) [inline misc/power.c:15]:
```suggestion
 * @returns the value {@code a}<sup>{@code b}</sup>.
```

**Comment by shellhub** [comment]
@Panquesito7 Done.

**Comment by shellhub** [comment]
@Panquesito7 I accepted your suggestion.

File: misc/power.c (hunk 5 above)
Comment (Panquesito7) [reply misc/power.c:4]: It is preferably to add the author's name.

File: misc/power.c [hunk 15]
```diff
@@ -0,0 +1,49 @@
+/**
+ * @file
+ * @brief Program to calculate
+ * [exponentiation](https://en.wikipedia.org/wiki/Exponentiation)
+ *
+ * @author [Du Yuanchao](https://github.com/shellhub)
+ *
+ * @author [Du Yuanchao](https://github.com/shellhub)
```
Comment (Panquesito7) [inline misc/power.c:8]:
```suggestion
 * @author [Du Yuanchao](https://github.com/shellhub)
```

File: misc/power_recursion.c (hunk 6 above)
Comment (Panquesito7) [inline misc/power_recursion.c:5]:
```suggestion
 * recursion algorithm.
 *
 * @author [Du Yuanchao](https://github.com/shellhub)
```

File: misc/power.c (hunk 15 above)
Comment (shellhub) [inline misc/power.c:8]:
```suggestion
```

**Review by shellhub — COMMENTED** [review]
delete duplicate

**Review by Panquesito7 — APPROVED** [review]
LGTM. 👍 😄

**Comment by shellhub** [comment]
@Panquesito7 please review again.

**Pull request merged**